traversal/walk.go

```diff
@@ -120,6 +120,25 @@ func (prog Progress) walkAdv(n datamodel.Node, s selector.Selector, fn AdvVisitF
 }
 
 func (prog Progress) walkAdv_iterateAll(n datamodel.Node, s selector.Selector, fn AdvVisitFn) error {
+	if rs, ok := s.(selector.Transformable); ok {
```

Comment:
in terms of defensive programming, using `s, ok := ...` here would make sure you only have one version of the typed selector in scope